returnn/torch/updater.py

```diff
@@ -122,8 +122,9 @@ def load_optimizer(self, filename):
         :param str filename: File from which to load the optimizer state.
         """
         print("Load optimizer %s" % filename, file=log.v4)
-        optimizer_state = torch.load(filename)
+        optimizer_state = torch.load(filename, map_location=torch.device("cpu"))
```

Comment:
Why? Why not `self._device`?

Reply:
is the optimiser also send to self._device? I couldn't find code that sends the optimiser to self._device

Reply:
It should. It would be very inefficient/slow if it is not. It would probably also throw an error, as you calculate gradients on GPU, the parameters are on GPU, you calculate the gradient updates on GPU, and that involves the optimizer state (momentum etc), so if that state is on CPU, you should get an error. So I'm pretty sure it is normally on GPU. I'm not sure what code exactly moves it to GPU. Maybe it is automatically on the same device as the parameters.